spec/models/credential_spec.rb

```diff
@@ -15,6 +15,22 @@
     credential
   end
 
+  it 'gets the user id from a JWT' do
+    # generate a keypair to build a sample JWT
+    rsa_private = OpenSSL::PKey::RSA.generate 2048
+    rsa_public = rsa_private.public_key
+
+    # encode a fake app state
+    contents = { data: {id: 3}, exp: (Time.now + 1.minute).to_i }
+    token = JWT.encode contents, rsa_private, 'RS512'
+
+    # use our public key for decryption instead of the default one
+    allow_any_instance_of(Panoptes::Client).to receive(:jwt_signing_public_key).and_return(rsa_public)
```

Comment:
I still think you could get away with just stubbing the client instance with a test double instead of creating the keys for signing tokens, etc.
```
# add the extra data payloads you need for this code path
client_double = double(current_user: { id: 3 }
allow_any_instance_of(Panoptes::Client).to receive(:new)and_return(client_double)
credential = Credential.create!(token: token)
expect(credential.current_user_id).to eq(3)
```

Reply:
i definitely could just stub that if it's better. i wanted to test the actual encoding/decoding since it's a bit sensitive to the format/contents of the token, but i'm happy to remove if you'd rather i not. this is going to be moved to the panoptes client in a pull request i'm working on right now.

Reply:
up to you, the code path your testing actually derives the test data from the client itself. If you trust the client is tested properly then there isn't really a need to test it's internals here. 

Nice to know the functionality is moving into the client :)

Reply:
yep, i'll be stubbing the tests exactly as you describe once i've moved the tests into the client. i'm hoping to banish the letters jwt completely from caesar and make the client responsible for all of it.